AI/VCAI/Pathfinding/Actions/BoatActions.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+* BoatActions.cpp, part of VCMI engine
+*
+* Authors: listed in file AUTHORS in main folder
+*
+* License: GNU General Public License v2.0 or later
+* Full text of license available in license.txt file, in main folder
+*
+*/
+
+#include "StdInc.h"
+#include "../../Goals/AdventureSpellCast.h"
+#include "../../Goals/BuildBoat.h"
+#include "../../../../lib/mapping/CMap.h"
+#include "../../../../lib/mapObjects/MapObjects.h"
+#include "BoatActions.h"
+
+namespace AIPathfinding
+{
+	Goals::TSubgoal BuildBoatAction::whatToDo(HeroPtr hero) const
```

Comment:
Copying `HeroPtr` is not that cheap. As long as all of this is changed, I`d like to see more usage of constant references.

Reply:
Fixed